src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -664,26 +669,23 @@ private static LibraryIncludeFlags GetLibraryIncludeFlags(string value, LibraryI
         /// <param name="entryProjectPath">The full path to the main project or solution file.</param>
         /// <param name="globalProperties">An <see cref="IDictionary{String,String}" /> representing the global properties for the project.</param>
         /// <returns></returns>
-        private List<ProjectGraphEntryPoint> GetProjectGraphEntryPoints(string entryProjectPath, IDictionary<string, string> globalProperties)
+        private static async ValueTask<List<ProjectGraphEntryPoint>> GetProjectGraphEntryPointsAsync(string entryProjectPath, IDictionary<string, string> globalProperties)
         {
             // If the project's extension is .sln, parse it as a Visual Studio solution and return the projects it contains
-            if (string.Equals(Path.GetExtension(entryProjectPath), ".sln", StringComparison.OrdinalIgnoreCase))
+            var extension = Path.GetExtension(entryProjectPath);
+            if (string.Equals(extension, ".sln", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(extension, ".slnx", StringComparison.OrdinalIgnoreCase))
             {
-                var solutionFile = SolutionFile.Parse(entryProjectPath);
+                ISolutionSerializer solutionSerializer = SolutionSerializers.GetSerializerByMoniker(entryProjectPath);
+                SolutionModel solution = await solutionSerializer.OpenAsync(entryProjectPath, CancellationToken.None);
 
-                IEnumerable<ProjectInSolution> projectsKnownToMSBuild = solutionFile.ProjectsInOrder.Where(i => i.ProjectType == SolutionProjectType.KnownToBeMSBuildFormat);
```

Comment:
@jeffkl the package that knows how to read/write slnx files doesn't seem to have any equivalent API to `KnownToBeMSBuildFormat`.   Any suggestions?

Reply:
This logic is custom in MSBuild: https://github.com/dotnet/msbuild/blob/82c3342f0d676370a735b47dfb984008697bf2e7/src/Build/Construction/Solution/SolutionFile.cs#L1535

That `SolutionFile` class should be able to parse `.slnx` files as of MSBuild 17.14 because of https://github.com/dotnet/msbuild/pull/10836.  I wonder if it would still work to just use this API?

Reply:
Yes, it does work!  Makes the changes made to our dotnet CLI commands unnecessary, but whatever.  This PR is much simpler now, and avoids the assembly load (or include the assembly in the vsix) question, which is great.